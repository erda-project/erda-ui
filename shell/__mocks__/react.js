// Copyright (c) 2021 Terminus, Inc.
//
// This program is free software: you can use, redistribute, and/or modify
// it under the terms of the GNU Affero General Public License, version 3
// or later ("AGPL"), as published by the Free Software Foundation.
//
// This program is distributed in the hope that it will be useful, but WITHOUT
// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
// FITNESS FOR A PARTICULAR PURPOSE.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program. If not, see <http://www.gnu.org/licenses/>.

const react = require('react');
// Resolution for requestAnimationFrame not supported in jest error :
// https://github.com/facebook/react/issues/9102#issuecomment-283873039
global.window = global;
window.addEventListener = () => {};
window.requestAnimationFrame = () => {
  // throw new Error('requestAnimationFrame is not supported in Node');
};

module.exports = react;
