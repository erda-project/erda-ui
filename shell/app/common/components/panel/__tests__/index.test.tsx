// Copyright (c) 2021 Terminus, Inc.
//
// This program is free software: you can use, redistribute, and/or modify
// it under the terms of the GNU Affero General Public License, version 3
// or later ("AGPL"), as published by the Free Software Foundation.
//
// This program is distributed in the hope that it will be useful, but WITHOUT
// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
// FITNESS FOR A PARTICULAR PURPOSE.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program. If not, see <http://www.gnu.org/licenses/>.

import React from 'react';
import { render } from '@testing-library/react';
import Panel, { PanelProps } from '../index';

const props: PanelProps = {
  isMultiColumn: false,
  fields: [
    {
      label: '测试label',
      valueKey: 'test-label',
      tips: 'test-tips',
    },
    {
      hide: true,
      label: '测试label-hide',
      valueKey: 'test-label-hide',
      tips: 'test-tips-hide',
    },
  ],
};

describe('Panel', () => {
  const mockFn = jest.fn();
  beforeEach(() => {
    jest.doMock('rc-resize-observer', () => {
      const ResizeObserverMock: React.FC<{ onResize: () => void }> = ({ onResize, children }) => {
        mockFn.mockImplementation(() => {
          onResize();
        });
        return <div className="mock-resize-observer">{children}</div>;
      };
      return ResizeObserverMock;
    });
  });
  it('should render well', () => {
    const wrapper = render(<Panel {...props} />);
    expect(wrapper.container.querySelectorAll('.erda-panel-label').length).toBe(1);
  });
  it('should render with multiColumn', () => {
    const wrapper = render(<Panel {...props} isMultiColumn />);
    expect(wrapper.container).toMatchSnapshot();
  });
});
