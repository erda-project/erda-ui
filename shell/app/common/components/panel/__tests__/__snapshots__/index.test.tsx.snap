// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel should render with multiColumn 1`] = `
<div>
  <div
    class="erda-panel"
  >
    <div
      class="ant-row"
      style="margin-left: -6px; margin-right: -6px;"
    />
  </div>
</div>
`;
